src/views/product-integrations-landing/components/searchable-integrations-list/index.tsx

```diff
@@ -42,6 +46,33 @@ export default function SearchableIntegrationsList({
 		}
 	)
 
+	/**
+	 * Track an "integration_library_searched" event when the filterQuery changes
+	 *
+	 * Note: we only want to track this event if the query input is meaningful.
+	 * We consider query input lengths of more than 2 characters to be meaningful
+	 * (in fact, we don't filter results unless the query is > 2 chars long).
+	 *
+	 * Note as well that other dimensions of filtering will fire events here.
+	 * For example, consider the following interactions:
+	 * - Visitor enters filterQuery "aws". This yields 8 results.
+	 *   We track "searched event" with { "aws", 8 }.
+	 * - Visitor also filters for "Config Sourcer" components.
+	 * 	 This new combination of "aws" + component filter yields 1 result.
+	 * 	 We track a "filter selected event" via `setComponentCheckedArray`.
+	 * - Since filteredIntegrations.length changes, "searched event" fires again.
+	 *   We track "searched event" with { "aws", 1 } (even though the visitor
+	 *   has not changed the `filterQuery`).
+	 */
+	useEffect(() => {
+		if (filterQuery.length > 2) {
+			integrationLibrarySearchedEvent({
+				search_query: filterQuery,
+				results_count: filteredIntegrations.length,
+			})
+		}
+	}, [filterQuery, filteredIntegrations])
```

Comment:
Oh also I missed your comment above - I see you acknowledge this!

Reply:
@zchsh @BrandonRomano in case it's helpful, here's [how `CommandBar`'s search command debounces](https://github.com/hashicorp/dev-portal/blob/main/src/components/command-bar/commands/search/components/dialog-body/index.tsx#L83-L92)!

```tsx
/**
 * Delay sending off search queries while the user is typing.
 */
useEffect(() => {
  const typingDebounce = setTimeout(() => {
    setSearchQuery(currentInputValue)
  }, 300)

  return () => clearTimeout(typingDebounce)
}, [currentInputValue])
```

I imagine we're probably close to ready to abstract for reuse? It seems useful here, in Tutorials Library (if it's not already happening), and of course is already in the search `CommandBar` command. I imagine something like this:

```tsx
/**
 * Delay invoking a callback until after a user has finished typing.
 */
const useTypingDebounce = ({
  callback,
  debounceTime = 300,
  inputValue,
}) => {
  useEffect(() => {
    const typingDebounce = setTimeout(() => {
      callback()
    }, debounceTime)

    return () => clearTimeout(typingDebounce)
  }, [inputValue])
}
```

What do y'all think?

Reply:
I do like that @ashleemboyer, something like that would do the trick ✨

Reply:
Thanks for this y'all 💖 I've added `useTypingDebounce` to `lib/hooks` - thanks for the huge head start on that @ashleemboyer 🙌